apps/documents/serializers.py

```diff
@@ -8,49 +24,25 @@
 from rest_framework_json_api import serializers
 from influxdb_metrics.loader import log_metric
 
-from apps.utils import UpperEnumField
-from .models import Document, DocumentType, FileType, UploadStatus, IMAGE_TYPES
+from apps.utils import UpperEnumField, S3PreSignedMixin
+from .models import Document, DocumentType, FileType, UploadStatus
 from .rpc import DocumentRPCClient
 
 LOG = logging.getLogger('transmission')
 
 
-class DocumentSerializer(EnumSupportSerializerMixin, serializers.ModelSerializer):
+class AbstractDocumentSerializer(S3PreSignedMixin, EnumSupportSerializerMixin, serializers.ModelSerializer):
     document_type = UpperEnumField(DocumentType, lenient=True, read_only=True, ints_as_names=True)
     file_type = UpperEnumField(FileType, lenient=True, read_only=True, ints_as_names=True)
     upload_status = UpperEnumField(UploadStatus, lenient=True, ints_as_names=True)
     presigned_s3 = serializers.SerializerMethodField()
 
-    class Meta:
-        model = Document
-        if settings.PROFILES_ENABLED:
-            exclude = ('owner_id',)
-        read_only_fields = ('shipment',)
+    def __init__(self, *args, **kwargs):
+        super(AbstractDocumentSerializer, self).__init__(*args, **kwargs)
+        self._s3_bucket = settings.DOCUMENT_MANAGEMENT_BUCKET
```

Comment:
If this is a property that is always set to a setting, I think we can move it out of the __init__ and just define it as a class property

Reply:
Got it :+1: